src/bp/core/services/dialog/prompt-manager.ts

```diff
@@ -258,6 +264,14 @@ export class PromptManager {
       }
     }
 
+    if (context.pastPromptCandidates) {
+      context.pastPromptCandidates = context.pastPromptCandidates
+        .map(c => ({ ...c, turns_ago: c.turns_ago++ }))
```

Comment:
just so you know ++ operator mutates the original object you don't have to map, create a new array, etc

```ts
context.pastPromptCandidates.foreach(c => c.turns_ago++).filter(c => c.turns_ago < 10)
```

Reply:
in fact ... you could do all this in a single lodash statement (even mutating context.pastPromptCandidates

```ts
candidates = _.chain(context.pastPromptCandidates)
 .forEach(c => c.turns_ago++)
 .filter(c => c.turns_ago < 10)
 .differenceBy(candidates, c => `${c.source}${c.value_raw}`)
 .value()
```

Reply:
Yes, but forEach is void so I can't use it to filter afterwards.